sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java

```diff
@@ -719,32 +724,38 @@ public DataLakeServiceVersion getServiceVersion() {
     }
 
     /**
-     * Takes in a destination path and creates a DataLakePathAsyncClient with a new path name
+     * Takes in a destination and creates a DataLakePathAsyncClient with a new path
+     * @param destinationFileSystem The destination file system
      * @param destinationPath The destination path
      * @return A DataLakePathAsyncClient
      */
-    DataLakePathAsyncClient getPathAsyncClient(String destinationPath) {
+    DataLakePathAsyncClient getPathAsyncClient(String destinationFileSystem, String destinationPath) {
+        if (CoreUtils.isNullOrEmpty(destinationFileSystem)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
```

Comment:
Any reason this doesn't default `destinationFileSystem` to the current file system? Looks like `renameWithResponse` will pass the current file system as the `destinationFileSystem` value. Seems odd to allow null in one case and not in another very similar case.

Reply:
Sure, I can change that